src/main/kotlin/no/nav/syfo/AccessTokenClient.kt

```diff
@@ -19,31 +21,37 @@ class AccessTokenClient(
         private val client: HttpClient
 ) {
     private val log: Logger = LoggerFactory.getLogger("no.nav.syfo.syfonarmesteleder")
+    private val mutex = Mutex()
+    @Volatile
+    private var token: AadAccessToken? = null
 
     suspend fun hentAccessToken(resource: String): String {
-        log.trace("Henter token fra Azure AD")
-        // TODO: Remove this workaround whenever ktor issue #1009 is fixed
-        val response: AadAccessToken = client.post<HttpResponse>(aadAccessTokenUrl) {
-            accept(ContentType.Application.Json)
-            method = HttpMethod.Post
-            body = FormDataContent(Parameters.build {
-                append("client_id", clientId)
-                append("resource", resource)
-                append("grant_type", "client_credentials")
-                append("client_secret", clientSecret)
-            })
-        }.use { it.call.response.receive<AadAccessToken>() }
-        log.trace("Har hentet accesstoken")
-        return response.access_token
+        val omToMinutter = Instant.now().plusSeconds(120L)
+        return mutex.withLock {
```

Comment:
Er mutexen her for at alle trådene ikke oppdatere token samtidig når det først er i ferd med å gå ut?

Reply:
Ja, og at kun én tråd kan aksessere tokenet...

Reply:
Smart 👍